src/lib/components/Login/OnaOauthLogin.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import './OnaOauthLogin.scss';
+import { useOAuthLogin } from '@onaio/gatekeeper/dist/components/login';
+import { AuthorizationGrantType } from '@onaio/gatekeeper';
+
+const mapStateToProps = state => {
+  const { APP } = state;
+  return {
+    appIcon: APP.appIcon,
+    loginIcon: APP.appLoginIcon,
+    publicUsername: APP.authPublicUsername,
+    publicPassword: APP.authPublicPassword,
+    clientID: APP.authClientID,
+    redirectUri: APP.authRedirectUri,
+  };
+};
+
+class OnaOauthLogin extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      authorizationUris: {},
+    };
+  }
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.clientID !== this.props.clientID ||
+      prevProps.redirectUri !== this.props.redirectUri
+    ) {
+      const providers = {
+        onadata: {
+          accessTokenUri: '',
+          authorizationUri: 'https://api.ona.io/o/authorize/',
+          clientId: this.props.clientID,
+          redirectUri: this.props.redirectUri,
+          scopes: ['read', 'write'],
+          state: 'abc',
+          userUri: 'https://api.ona.io/user',
+        },
+      };
+      const options = {
+        providers,
+        authorizationGrantType: AuthorizationGrantType.IMPLICIT,
+      };
+      const authorizationUris = useOAuthLogin(options);
+      this.setState({
+        authorizationUris,
+      });
+    }
+  }
+
+  getProviderKey() {
+    return Object.keys(this.state.authorizationUris).filter(key => key === 'onadata')[0];
+  }
+
+  render() {
+    const { loginIcon, appIcon, publicPassword, publicUsername } = this.props;
+    const loginImage = {
+      background: `url(${loginIcon || appIcon}) no-repeat center center`,
+    };
+    const providerKey = this.getProviderKey();
+
+    return (
+      <div className="login">
+        <div>
+          <form className="login-form">
+            <div className="form-group">
+              <div className="brand-login" style={loginImage}></div>
+              {publicPassword && publicUsername ? (
+                <div>
+                  <small>Username: {publicUsername}</small>
+                  <br />
+                  <small>Password: {publicPassword}</small>
+                </div>
+              ) : null}
```

Comment:
@cKellyDesign I agree. This was client-driven. I think at some point discover didn't support public dashboards and accessing this on an iframe resulted to csrf errors.  We decided to implement oauth to login users once on all dashboards. We had thought of having a landing page that offered two opti

Reply:
Thanks for the intel @kahama94 - I think with the updates to the Superset Public role Discover should support public dashboards, right?

Reply:
@cKellyDesign Discover currently supports public dashboards this is really nice since we can now have public sites with no oauth and private one's with oauth. We can do away with having oauth as a must have for all sites but rather for one's which are private. :clap: